### PR TITLE
K8SPS-16: Create MySQL primary service

### PR DESCRIPTION
**Pull request by egegunes**

### CODE REVIEW

File: pkg/database/orchestrator/client/client.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	return &orcClient{
+		host:   host,
+		client: c,
+	}
+}
+
+func (o *orcClient) ClusterMaster(clusterHint string) (*clusterMasterResp, error) {
+	req, err := http.NewRequest(http.MethodGet, o.host+"/api/master/"+clusterHint, nil)
```
[the remark this answers is not in the document]
Comment (egegunes) [reply pkg/database/orchestrator/client/client.go:70]: https://jira.percona.com/browse/K8SPS-19

File: pkg/api/v2/perconaserverformysql_types.go [hunk 2]
```diff
@@ -264,13 +268,35 @@ func (v *VolumeSpec) reconcile() {
...
-		"app.kubernetes.io/part-of":    "percona-server",
+		NameLabel:      "percona-server",
+		InstanceLabel:  cr.Name,
+		ManagedByLabel: "percona-server-operator",
+		PartOfLabel:    "percona-server",
+	}
+}
+
+func (cr *PerconaServerForMySQL) ClusterHint() string {
+	return fmt.Sprintf("%s.%s", cr.Name, cr.Namespace)
```
[the remark this answers is not in the document]
Comment (egegunes) [reply pkg/api/v2/perconaserverformysql_types.go:290]: Is it OK for you to do it in a separate PR?

File: pkg/database/mysql/service.go [hunk 3]
```diff
@@ -3,12 +3,18 @@ package mysql
...
+
+	v2 "github.com/percona/percona-server-mysql-operator/pkg/api/v2"
 )
 
 func (m *MySQL) ServiceName() string {
 	return m.Name()
 }
 
+func (m *MySQL) PrimaryServiceName() string {
+	return m.Name() + "-primary"
```
[the remark this answers is not in the document]
Comment (egegunes) [reply pkg/database/mysql/service.go:15]: Asked in Slack

File: pkg/pod/reconcile.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
+package pod
```
[the remark this answers is not in the document]
Comment (egegunes) [reply pkg/pod/reconcile.go]: https://github.com/percona/percona-server-mysql-operator/pull/11/commits/2c3065f7060543ba221b11ac93bbb10f4897e9e0

File: pkg/database/orchestrator/client/client.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
+package client
```
[the remark this answers is not in the document]
Comment (egegunes) [reply pkg/database/orchestrator/client/client.go:1]: https://github.com/percona/percona-server-mysql-operator/pull/11/commits/2c3065f7060543ba221b11ac93bbb10f4897e9e0

Unfortunately there are some places we need to use `master`. For example orchestrator has the endpoint `/api/master` and we're using it. Another one is `MasterKey` field in the `instance` type to match the field name in orchestrator's response.

File: pkg/api/v2/perconaserverformysql_types.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (delgod) [reply pkg/api/v2/perconaserverformysql_types.go:291]: please create ticket

[the remark this answers is not in the document]
Comment (egegunes) [reply pkg/api/v2/perconaserverformysql_types.go:291]: https://jira.percona.com/browse/K8SPS-21

File: pkg/database/mysql/service.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (delgod) [reply pkg/database/mysql/service.go:15]: please create ticket

**Review by delgod — APPROVED** [review]

File: pkg/database/mysql/service.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (egegunes) [reply pkg/database/mysql/service.go:15]: https://jira.percona.com/browse/K8SPS-22

File: pkg/database/mysql/mysql.go [hunk 6]
```diff
@@ -36,3 +37,9 @@ func (m *MySQL) Name() string {
 func (m *MySQL) Namespace() string {
 	return m.cluster.Namespace
 }
+
+func IsMySQL(obj client.Object) bool {
```
Comment (hors) [inline pkg/database/mysql/mysql.go:41]: This code is not used but we discussed it with @egegunes  and it will be removed in another PR.

**Review by hors — APPROVED** [review]
